src/EnergyPlus/SystemReports.cc

```diff
@@ -488,40 +404,34 @@ void InitEnergyReports(EnergyPlusData &state)
                     }
                 }
 
+                int EquipNum = 0;
+                int SubEquipNum = 0;
+                int SubSubEquipNum = 0;
+                int CompNum = 0;
+                int SubCompNum = 0;
+                int SubSubCompNum = 0;
                 // Eliminate duplicates in the connection arrays
                 if (allocated(state.dataAirSystemsData->ZoneCompToPlant)) {
                     EquipNum = isize(state.dataAirSystemsData->ZoneCompToPlant);
-                } else {
-                    EquipNum = 0;
                 }
```

Comment:
I don't know the answer so I will ask if this works the same.

int EquipNum = state.dataAirSystemsData->ZoneCompToPlant.size();